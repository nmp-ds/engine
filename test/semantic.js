import { setup } from './_helpers.js';
import { expect, test } from 'vitest';

setup();

test('it generates css based on semantic warp tokens', async ({ uno }) => {
  const classes = [
    's-bg-default',
    's-bg-positive-active-hover',
    's-text-default',
    's-text-link-active',
    's-border-default',
    's-border-l-disabled',
    's-border-r-primary-active-hover',
    's-border-t-positive-default',
    's-border-b-negative-hover',
    's-border-x-warning-active',
    's-border-y-info-subtle-active-hover',
  ];
  const antiClasses = [
    's-background-default',
    's-background-positive-active-hover',
    's-font-default',
    's-font-link-active',
    's-bordercolor-default',
    's-bordercolor-l-disabled',
  ];
  const { css } = await uno.generate([...classes, ...antiClasses]);
  expect(css).toMatchSnapshot();
});
