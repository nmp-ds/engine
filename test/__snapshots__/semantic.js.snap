// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it generates css based on semantic warp tokens 1`] = `
"/* layer: default */
.s-bg-default{background-color:var(--w-s-color-background-default);}
.s-bg-positive-active-hover{background-color:var(--w-s-color-background-positive-active-hover);}
.s-text-default{color:var(--w-s-color-text-default);}
.s-text-link-active{color:var(--w-s-color-text-link-active);}
.s-border-default{border-color:var(--w-s-color-border-default);}
.s-border-b-negative-hover{border-bottom-color:var(--w-s-color-border-negative-hover);}
.s-border-l-disabled{border-left-color:var(--w-s-color-border-disabled);}
.s-border-r-primary-active-hover{border-right-color:var(--w-s-color-border-primary-active-hover);}
.s-border-t-positive-default{border-top-color:var(--w-s-color-border-positive-default);}
.s-border-x-warning-active{border-left-color:var(--w-s-color-border-warning-active);border-right-color:var(--w-s-color-border-warning-active);}
.s-border-y-info-subtle-active-hover{border-top-color:var(--w-s-color-border-info-subtle-active-hover);border-bottom-color:var(--w-s-color-border-info-subtle-active-hover);}"
`;
